src/components/data/Tops/TopArtists/TopArtists.tsx

```diff
@@ -1,31 +1,50 @@
-import { Stack, StackDivider, Text, Button } from '@chakra-ui/react'
+import { Stack, StackDivider, Text, Button, Select } from '@chakra-ui/react'
 import { useTranslation } from 'react-i18next'
 import { useState } from 'react'
 
-import { useTopArtistsMedTerm } from 'api/spotify/personalization'
+import { useTopArtistsLongTerm, useTopArtistsMediumTerm, useTopArtistsShortTerm } from 'api/spotify/personalization'
 import { TopArtistArea } from './TopArtistArea'
 
 enum Mode { MORE, LESS }
 
 const TopArtists = (): JSX.Element => {
-  const { data } = useTopArtistsMedTerm()
+  const { data: longTermData } = useTopArtistsLongTerm()
+  const { data: mediumTermData } = useTopArtistsMediumTerm()
+  const { data: shortTermData } = useTopArtistsShortTerm()
+  const [data, setData] = useState(shortTermData)
   const { t } = useTranslation()
   const [itemsToShow, setItemsToShow] = useState(8)
   const [mode, setMode] = useState<Mode>(Mode.LESS)
 
+  const handleChange = (e :React.ChangeEvent<HTMLSelectElement>) => {
+    switch (e.target.value) {
+      case 's': setData(shortTermData); break
+      case 'm': setData(mediumTermData); break
+      case 'l': setData(longTermData); break
+      default: break
+    }
+  }
```

Comment:
tak jak ci pisałem, tutaj powinno być tylko ustawianie stanu jaki aktualnie timeRange jest wybrany, a następnie ten switch  opakowany w `useMemo`, tak na prawdę. Jeżeli zrezygnujemy z osobnego query dla każdego z timeRange to mozemy przechowywać jako `value` w `select`  odpowiedni string odpowiadający danej wartości przekazywanej do query to nawet to `useMemo` nie będzie potrzebne, a query powinno się przeładować (trzeba będzie najprawdopodbniej skorzystać z opcji `enabled` useQuery).